analyses/cell-type-ewings/exploratory_analysis/04-infercnv.Rmd

```diff
@@ -0,0 +1,480 @@
+
+```{r}
+# make df for plotting
+coldata_df <- sce |> 
+  scuttle::makePerCellDF(use.dimred = "UMAP") |> 
+  # replace UMAP.1 with UMAP1
+  dplyr::rename_with(
+        \(x) stringr::str_replace(x, "^UMAP\\.", "UMAP")
+      ) |> 
+  # add in per cell cnv information 
+  dplyr::left_join(cnv_metadata_df, by = c("barcodes"))
+
+# remove MT and GL chr, comments in docs recommend that these are not useful and can be inaccurate 
+chr_columns_to_keep <- which(!stringr::str_detect(names(coldata_df), "chrMT|chrGL000219.1|chrGL000194.1"))
+all_cnv_df <- coldata_df[, chr_columns_to_keep]
+```
+
+The metadata that we added includes the following columns for each chromosome: 
+
+- `has_cnv_chr1`
+- `has_loss_chr1`
+- `has_dupli_chr1`
+- `proportion_cnv_chr1`
+- `proportion_loss_chr1`
+- `proportion_dupli_chr1`
+- `proportion_scaled_cnv_chr1`
+- `proportion_scaled_loss_chr1`
+- `proportion_scaled_dupli_chr1`
+
+The `has` columns are boolean columns indicating if that cell has presence of a CNV, loss, or duplication event. 
+If `has_cnv` is TRUE if either `has_loss` or `has_dupli` is TRUE. 
+
+See [the comments in this issue](https://github.com/broadinstitute/infercnv/issues/208) explaining what the contents of the columns are. 
+
+One approach to classifying tumor cells is to look for cells that have a lot of CNVs. 
+Let's start by just summing up the total number of chromosomes with CNVs per cell and plotting that. 
+
+
+```{r}
+# grab any columns that have has_cnv 
+has_cnv_cols <- names(all_cnv_df)[startsWith(colnames(all_cnv_df), "has_cnv")]
+
+# don't include the normal cells that were provided as reference in scaling 
+cnv_df <- all_cnv_df |>
+  # get total number of chromosomes with cnvs per cell and then scale that number 
+  dplyr::mutate(
+    has_cnv = ifelse(
+      stringr::str_detect(subcluster, "reference"), 
+      NA_real_, 
+      rowSums(across(has_cnv_cols))
+    ),
+    z_scaled_cnv = scale(has_cnv))
+```
+
+
+```{r}
+# plot the total number of CNVs
+ggplot(cnv_df, aes(x = UMAP1, y= UMAP2, color = has_cnv)) +
+  geom_point(alpha = 0.5, size = 0.5) +
+  theme_bw() +
+  scale_color_viridis_c()
+```
+This shows that the total CNVs are highest in the group of cells that correspond to the same cells that are manually annotated as tumor cells, which is promising. 
+Those cells in grey (bottom left) are cells used as a reference. 
+If we were to use total CNVs we would need some sort of cut off for the number. 
+Let's look at the distribution of total CNVs and also the scaled total CNVs to see if we can identify a cut off. 
+
+```{r}
+ggplot(cnv_df, aes(x = has_cnv)) +
+  geom_density() +
+  theme_bw() +
+  labs(x = "Total number of CNVs")
+
+ggplot(cnv_df, aes(x= z_scaled_cnv)) + 
+  geom_density() +
+  theme_bw() +
+  labs(x = "z-scaled chromosome CNV count")
+```
+
+Looking at the scaled number of CNVs it does look like we could use 0 as the cutoff and call any cell that has > 0 CNVs (after scaling) as tumor and all other cells as normal. 
```

Comment:
This is the equivalent of just using the mean as the cutoff, so let's make that explicit?